diff --git a/build/gulpfile.vscode.ts b/build/gulpfile.vscode.ts
--- a/build/gulpfile.vscode.ts
+++ b/build/gulpfile.vscode.ts
@@ -602,3 +602,3 @@ BUILD_TARGETS.forEach(buildTarget => {
 		);
-		const vscodeTask = task.define(`vscode${dashed(platform)}${dashed(arch)}${dashed(minified)}`, task.series(
+		const prepackTask = task.define(`vscode${dashed(minified)}-prepack`, task.series(
 			copyCodiconsTask,
@@ -608,2 +608,6 @@ BUILD_TARGETS.forEach(buildTarget => {
 			compileExtensionMediaBuildTask,
+		));
+		task.task(prepackTask);
+
+		const packingTask = task.define(`vscode${dashed(platform)}${dashed(arch)}${dashed(minified)}-packing`, task.series(
 			writeISODate('out-build'),
@@ -612,2 +616,8 @@ BUILD_TARGETS.forEach(buildTarget => {
 		));
+		task.task(packingTask);
+
+		const vscodeTask = task.define(`vscode${dashed(platform)}${dashed(arch)}${dashed(minified)}`, task.series(
+			prepackTask,
+			packingTask,
+		));
 		task.task(vscodeTask);
